### PR TITLE
bump version, merge develop, trigger release process

### PR DESCRIPTION
**Pull request by sunchunqiang**
*Issue #, if available:*

*Description of changes:*

The changes includes
1. bump android sdk version from circleci
2. merge develop to master from circleci when there is a commit to develop
3. start release process when there is bump android sdk version commit

### CODE REVIEW

File: CircleciScripts/bump_sdk_version.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        bump_pomxml(pomfile, newsdkversion, newasfversion)
+
+
+#define which files whose version number should be replaced and how to replace
+replaces = [
+    {
+        "match" : 'fabric-version=.*', 
+        "replace" : 'fabric-version=[version]',
+        "files" : [
+            "aws-android-sdk-cognito/src/main/resources/fabric/com.amazonaws.aws-android-sdk-cognito.properties"
```
Comment (minbi) [inline CircleciScripts/bump_sdk_version.py]: There should be a similar file for core

File: CircleciScripts/bump_sdk_version.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        ]       
+    } ,
+    {
+        "match" : 'private static volatile String version = ".*";', 
+        "replace" : 'private static volatile String version = "[version]";',
+        "files" : [
+            "aws-android-sdk-core/src/main/java/com/amazonaws/util/VersionInfoUtils.java"
```
Comment (minbi) [inline CircleciScripts/bump_sdk_version.py]: I believe there is a unit test for this version as well

File: .circleci/config.yml [hunk 3]
```diff
@@ -477,20 +546,31 @@ jobs:
     environment:
       JVM_OPTS: -Xmx1024m
     steps:
+      - skip_job_if_required
```
Comment (minbi) [inline .circleci/config.yml]: skip_job_if_overriden

File: .circleci/config.yml [hunk 4]
```diff
@@ -163,7 +176,53 @@ commands:
...
+              fi    
+  skip_job_if_required:
+    description: >-
+      skip the test job for ciecleci script test
+    steps:
+      - run:
+          name: checkifjobcanskip 
+          command: |
+            echo "check if job can be skipped"
+            # # we can add script here to halt a job for circleci script test
```
Comment (minbi) [inline .circleci/config.yml]: why is this all commented? do we not run any commands for this?

File: .circleci/config.yml [hunk 5]
```diff
@@ -507,14 +587,10 @@ jobs:
...
-          name: checkiftestcanskip 
-          command: |
-              if [ "${SKIP_INTEGRATIONTEST}" == "TRUE" ]; then
-                echo "project enviroment variabe SKIP_INTEGRATIONTEST is set with TRUE, skip all integration tests"
-                circleci step halt
-              fi          
+      - skip_job_if_required
+      - override_test_job
       - checkout
+      - skip_job_if_triggered_by_circleci
```
Comment (minbi) [inline .circleci/config.yml]: can this happen before checkout?

File: .circleci/config.yml [hunk 6]
```diff
@@ -566,7 +643,10 @@ jobs:
     environment:
       JVM_OPTS: -Xmx3200m
     steps:
+      - skip_job_if_required
+      - override_test_job      
       - checkout
+      - skip_job_if_triggered_by_circleci
```
Comment (minbi) [inline .circleci/config.yml]: same

File: .circleci/config.yml [hunk 7]
```diff
@@ -163,7 +176,53 @@ commands:
...
+      skip the test job is project variabe OVERRIDE_TEST is set with TRUE
+    steps:
+      - run:
+          name: checkiftestcanoverrided 
+          command: |
+              if [ "${OVERRIDE_TEST}" == "TRUE" ]; then
+                echo "project enviroment variabe OVERRIDE_TEST is set with TRUE, skip all tests"
+                circleci step halt
+              fi    
+  skip_job_if_required:
```
Comment (minbi) [inline .circleci/config.yml]: skip job unless required

File: .circleci/config.yml [hunk 8]
```diff
@@ -477,20 +546,31 @@ jobs:
...
+      - skip_job_if_required
       - run:
-          name: check environment variabe 
+          name: reset OVERRIDE_TEST environment variabe 
           command: |
-              if [ "${SKIP_INTEGRATIONTEST}" == "TRUE" ]; then
-                echo "project enviroment variabe SKIP_INTEGRATIONTEST should not be TRUE at the moment, this will skip all integration tests"
-                exit 1
-              fi           
+            parameters='{"name":"OVERRIDE_TEST","value":"FALSE"}'
```
Comment (minbi) [inline .circleci/config.yml]: use first 7 chars of commit hash in variable name

File: CircleciScripts/bump_sdk_version.py (hunk 1 above)
Comment (sunchunqiang) [reply CircleciScripts/bump_sdk_version.py]: done

File: CircleciScripts/bump_sdk_version.py (hunk 2 above)
Comment (sunchunqiang) [reply CircleciScripts/bump_sdk_version.py]: done

File: .circleci/config.yml (hunk 7 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 8 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

**Pull request merged**